lib/rendition.js

```diff
@@ -42,17 +102,158 @@ class Rendition {
+        // explicitly set mimetype would get priority
+        if(mime){ // to handle case when this._mime is set to default DEFAULT_CONTENT_TYPE
+            this._mime = mime;
+        }
+
+        // explicitly set encoding would get priority
+        if(encoding){ // to handle case when this.encoding would have a default value
+            this._encoding = encoding;
+        }
+
+        // shellscript mostly ------------------------------------------------------
+        if(this.mimeInfoPath && await fs.pathExists(this.mimeInfoPath)){ // use the path containing the file with mime information
+            let mimeInfoContent = await fs.readFile(this.mimeInfoPath);
+            mimeInfoContent = mimeInfoContent.toString();
+            const mimeInfo = mimeInfoContent.split(";");
+
+            if(mimeInfo && mimeInfo.length === 2){ // we got something in an expected format
+                this._mime = mime || mimeInfo[0].trim() || DEFAULT_CONTENT_TYPE;
+                this._encoding = encoding || mimeInfo[1].trim();
+            }
+        } 
+        // fallback if things are missing ------------------------------------------------------
+        if(!this._encoding || !this._mime){ // fallback to using file-type lib (won't set encoding)
+            try { 
+                await commandExists('file'); // throws if `file` is not found
+                const fileCmdResult = await getMimeInformationFromCommand(this);
+                this._mime = this._mime || fileCmdResult.mime;
+                this._encoding = this._encoding || fileCmdResult.encoding;
+            } catch(err){
+                console.log('`file` command not found', err); // err should be empty
+                console.log('Trying to find mimetype based on rendition file analysis (magic numbers)');
+                const filetypeInfo = await FileType.fromFile(this.path);
+                if(filetypeInfo) { // is only set if something was recognized
+                    this._mime = this._mime || filetypeInfo.mime;
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns just the mimetype
+     */
+    async mimeType() {
+        if(!this._mime){
+            await this.setMimeInformation();
+        }
+        return this._mime;
+    }
+
+    /**
+     * Returns just the encoding
+     */
+    async encoding() {
+        if(!this._encoding){
+            await this.setMimeInformation();
+        }
+
+        if(this._encoding === "charset=binary"){
+            return null;
+        }
+        return this._encoding;
+    }
+
+    /**
+     * Returns a valid contenttype:
+     *      - plain mimetype for binary files
+     *      - mime-type + encoding for other (encoding for txt files)
+     */
+    async contentType() {
+        let contentType;
+
+        const encoding = await this.encoding();
+        const mimetype = await this.mimeType();
+
+        if(encoding){
+            contentType = `${mimetype}; ${encoding}`; 
```

Comment:
This must be `charset=${encoding}`

The encoding is just the encoding, not the string `charset=<encoding>`.

This is needed for:
1. Metadata.ENCODING must not include `charset`
2. The client when calling `setContentType()` would do:

```
setContentType("text/html", "utf-8")
```

not this
```
setContentType("text/html", "charset=utf-8")
```

This means parsing it from the `file` command result, but that's ok.